lib/server.js

```diff
@@ -48,6 +49,21 @@ function createServer (options, httpHandler) {
     } else {
       listenOptions.cb = cb
     }
+    if (listenOptions.signal) {
+      if (typeof listenOptions.signal.on !== 'function' && typeof listenOptions.signal.addEventListener !== 'function') {
+        throw new FST_ERR_LISTEN_OPTIONS_INVALID('Invalid options.signal')
+      }
+
+      if (listenOptions.signal.aborted) {
+        this.close()
```

Comment:
It seems like when `signal` is `aborted` you called `close` and it would still `listen` afterward?
Shouldn't we terminate the code execution after `this.close` here?

Reply:
It wont start because of this part - https://github.com/fastify/fastify/blob/8f9f3677d7bd3682974a75b060e6d0aeca809091/lib/server.js#L212
@Uzlopak Do you think it will be better to just terminate the function?

Reply:
I see why it would be fine because you added an extra if inside the `listenCallback`.
I am not sure if an extra if is better than early termination?